rust/azure_iot_operations_mqtt/src/session/wrapper.rs

```diff
@@ -48,19 +48,31 @@ pub struct SessionPubReceiver(managed_client::SessionPubReceiver);
 
 /// Options for configuring a new [`Session`]
 #[derive(Builder)]
-#[builder(pattern = "owned")]
+#[builder(pattern = "owned", build_fn(validate = "Self::validate"))]
 pub struct SessionOptions {
     /// MQTT Connection Settings for configuring the [`Session`]
-    pub connection_settings: MqttConnectionSettings,
+    connection_settings: MqttConnectionSettings,
     /// Reconnect Policy to by used by the `Session`
     #[builder(default = "Box::new(ExponentialBackoffWithJitter::default())")]
-    pub reconnect_policy: Box<dyn ReconnectPolicy>,
+    reconnect_policy: Box<dyn ReconnectPolicy>,
     /// Maximum number of queued outgoing messages not yet accepted by the MQTT Session
     #[builder(default = "100")]
-    pub outgoing_max: usize,
+    outgoing_max: usize,
     /// Indicates if the Session should use features specific for use with the AIO MQTT Broker
     #[builder(default = "true")]
-    pub aio_broker_features: bool,
+    aio_broker_features: bool,
+    /// Indicates if the Session should use AIO persistence
+    #[builder(default = "false")]
+    aio_persistence: bool,
+}
+
+impl SessionOptionsBuilder {
+    fn validate(&self) -> Result<(), String> {
+        if self.aio_persistence == Some(true) && self.aio_broker_features == Some(false) {
```

Comment:
This works, but perhaps broker_features should be it's own struct. Before, there was just the one broker feature, now perhaps there should be an optional struct. Would reduce validation and all these knock-on effects of it.